lib/sisimai/bite/email/sendmail.rb

```diff
@@ -0,0 +1,290 @@
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+        return nil unless mhead['subject'] =~ Re0[:subject]
+
+        unless mhead['subject'] =~ /\A[ \t]*Fwd?:/i
+          # Fwd: Returned mail: see transcript for details
+          # Do not execute this code if the bounce mail is a forwarded message.
+          return nil unless mhead['from'] =~ Re0[:from]
+        end
+
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
+        havepassed = ['']
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
+        blanklines = 0      # (Integer) The number of blank lines
+        readcursor = 0      # (Integer) Points the current cursor position
+        recipients = 0      # (Integer) The number of 'Final-Recipient' header
+        commandtxt = ''     # (String) SMTP Command name begin with the string '>>>'
+        esmtpreply = ''     # (String) Reply from remote server on SMTP session
+        sessionerr = false  # (Boolean) Flag, "true" if it is SMTP session error
+        connvalues = 0      # (Integer) Flag, 1 if all the value of $connheader have been set
+        connheader = {
+          'date'  => '',    # The value of Arrival-Date header
+          'rhost' => '',    # The value of Reporting-MTA header
+        }
+        anotherset = {}     # Another error information
+        v = nil
+
+        hasdivided.each do |e|
+          # Save the current line for the next loop
+          havepassed << e
+          p = havepassed[-2]
+
+          if readcursor.zero?
+            # Beginning of the bounce message or delivery status part
+            if e =~ Re1[:begin]
+              readcursor |= Indicators[:deliverystatus]
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] == 0
+            # Beginning of the original message part
+            if e =~ Re1[:rfc822]
+              readcursor |= Indicators[:'message-rfc822']
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] > 0
+            # After "message/rfc822"
+            if e.empty?
+              blanklines += 1
+              break if blanklines > 1
+              next
+            end
+            rfc822list << e
+
+          else
+            # Before "message/rfc822"
+            next if readcursor & Indicators[:deliverystatus] == 0
+            next if e.empty?
+
+            if connvalues == connheader.keys.size
+              # Final-Recipient: RFC822; userunknown@example.jp
+              # X-Actual-Recipient: RFC822; kijitora@example.co.jp
+              # Action: failed
+              # Status: 5.1.1
+              # Remote-MTA: DNS; mx.example.jp
+              # Diagnostic-Code: SMTP; 550 5.1.1 <userunknown@example.jp>... User Unknown
+              # Last-Attempt-Date: Fri, 14 Feb 2014 12:30:08 -0500
+              v = dscontents[-1]
```

Comment:
__[rubocop]__
* Style/NumericPredicate: Use `(readcursor & Indicators[:deliverystatus]).zero?` instead of `readcursor & Indicators[:deliverystatus] == 0`. [:ref](https://github.com/bbatsov/ruby-style-guide#predicate-methods)

<!-- Comment by [SideCI](https://sideci.com) -->